lib/min_heap.rb

```diff
@@ -14,18 +14,32 @@ def initialize
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(log n)
+  # Space Complexity: O(1)
   def add(key, value = key)
```

Comment:
👍 However due to the recursion it performs your heap_up method adds O(log n) space complexity due to the call stack.